File: Data Structures & Algorithms/validate-parentheses/submission-1.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []
        op = ['(', '{', '[']
        cl = [')', '}', ']']
        for ch in s:
            if ch in op:
                stack.append(ch)
            elif ch in cl:
                if len(stack) == 0:
                    return False
                opener = stack.pop()
                if ((opener == '(' and ch == ')') or
                   (opener == '{' and ch == '}') or
                   (opener == '[' and ch == ']')):
                    continue
                else:
                    return False
            else:
                return False
        if len(stack) != 0:
            return False
        return True

File: Data Structures & Algorithms/evaluate-reverse-polish-notation/submission-1.cpp
class Solution {
public:
    int evalRPN(vector<string>& tokens) {
        std::stack<int> eval;
        string num;
        string operators = "+-*/";
        int res;
        for (auto t : tokens) {
            if (!operators.contains(t)) {
                int val = std::stoi(t);
                eval.push(val);
            } else {
                int op2 = eval.top(); eval.pop();
                int op1 = eval.top(); eval.pop();
                if (t == "+") {
                    res = op1 + op2;
                } else if(t == "-") {
                    res = op1 - op2;
                } else if(t == "*") {
                    res = op1 * op2;
                } else {
                    res = op1 / op2;
                }
                eval.push(res);
            }
        }
        return eval.top();
    }
};
